### ISSUE TITLE
Proposal: Add helper function to apply a SAE or SAEDict to a model 

### ISSUE THREAD

**Issue by chanind** [jbloomAus/SAELens#84]
One of the most common use-cases for SAELens is to apply a pre-trained SAE visible on neuronpedia to a model to analyze activations. Currently we don't provide any helper to accomplish this - the user has to extract the hook point from the SAE, run the model, collect activations, and apply the SAE to the collected activations. Since this is likely a very common use-case we should make this super easy to do.

One possibility would be to have a context manager on `SparseAutoencoder` and `SparseAutoencoderDict`, enabling usage like:

```python
with sae.track_activations(model) as sae_cache:
    model(...)

inputs = sae_cache.inputs
feature_acts = sae_cache.feature_activations
outputs = sae_cache.outputs

with sae_dict.track_activations(model) as sae_dict_cache:
    model(...)

# dict outputs are the same as above, except in a dict for each SAE
sae1_inputs = sae_dict_cache['sae1_key'].inputs
...
```

Alternatively, we could add a `apply_and_run_with_cache()` method which both attaches the SAE and calls `run_with_cache()` on the underlying model, returning both the model cache and SAE cache

```python
model_cache, sae_cache = sae.apply_and_run_with_cache(model, ...)

# likewise, for SAEDict, return a dict of SAE caches
model_cache, sae_dict_cache = sae_dict.apply_and_run_with_cache(model, ...)
```

Some other things to think about:
- Do we want a separate method to inject the SAE into the model computation vs just tracking activations on the side?
- If we have a method to inject the SAE into model computation, should we have a way of adding in the SAE error term so we can attribute gradients to the SAE vs error term while ensuring the model computation is unchanged?
- Should we just always inject the SAE into model comptuation + error term by default, so the SAE gets gradients, tracks features, while not modifying model output?

**Comment by dtch1997** [comment]
#81 seems to accomplish this. I inject the SAE into the model and save its features and activations. Possibly that could be re-written to use something similar to an `ActivationCache` from TransformerLens.

**Comment by jbloomAus** [comment]
HookedSAE Transformer provides this functionality and was merged as part of the 3.0 release.
